### ISSUE TITLE
Add Grid for Canada

### ISSUE THREAD

**Issue by pciavarella** [NaturalGIS/ntv2_transformations#13]
The Canadian grid can be downloaded from here:

http://webapp.geod.nrcan.gc.ca/Files/NTV2_0.GSB

**Comment by gioman** [comment]
@pciavarella is the grid redistributable? with what license?

**Comment by gioman** [comment]
@pciavarella what are the tranformations this grid allows (EPSG codes please). Can you attach sample data?

**Comment by pciavarella** [comment]
Hi Giovanni,

Here are the transforms that we typically use:

From  (NAD27)                                               To (NAD83)
EPSG 4267                                                       EPSG 4269
EPSG 26710                                                     EPSG 26910
EPSG  26711                                                    EPSG 26911
EPSG  26712                                                    EPSG 26912

**Comment by pciavarella** [comment]
Hi Giovanni,

It`s distributed under the Open Government License – Canada

http://open.canada.ca/en/open-government-licence-canada

Peter Ciavarella

**Comment by gioman** [comment]
@pciavarella thanks! I'll have a look as soon as possible.

**Comment by saultdon** [comment]
Here's the updated open data link

http://open.canada.ca/data/en/dataset/b3534942-31ea-59cf-bcc3-f8dc4875081a

**Comment by mattman30** [comment]
Hi Giovanni,

We could really use the transformations that pciavarella outlined.  NAD27 to NAD83, please.

**Comment by gioman** [comment]
@mattman30  @pciavarella will try to gave it a look asap. Can you confirm the grids are still available at the provided addresses?

**Comment by mattman30** [comment]
Yes the link is still active.  More specifically, the download link from the webpage:
http://webapp.geod.nrcan.gc.ca/Files/NTV2_0.GSB
And the license link:
http://open.canada.ca/en/open-government-licence-canada

Thank you!

**Comment by gioman** [comment]
> Yes the link is still active. More specifically, the download link from the webpage:
> http://webapp.geod.nrcan.gc.ca/Files/NTV2_0.GSB
> And the license link:
> http://open.canada.ca/en/open-government-licence-canada
> 
> Thank you!

@mattman30 sorry I'm not fluent on north america CRSs...what are the EPSG coode of NAD27 and  NAD83 supported by those grids?

**Comment by tveinot** [comment]
Hi;
I am trying to transform NAD27 Double Stereographic PEI Imperial into NAD83 CSRS 2010 I have a CRS definition for the local NAD27 grid and the gsb file used to transform those grids.
Useful to you?
Tyler

**Comment by gioman** [comment]
> Useful to you?

if the gsb file is freely available and redistributable then yes, otherwise no.

**Comment by tveinot** [comment]
The CRS Description can be found here; http://www.nsgc.gov.ns.ca/todd/proj/pesi1.htm
I used it to make a PROJ4, the false eastings and northing are in feet, to make the CRS I had to convert them to metres.
I am checking on if there is a license for the gsb now, all I did was ask for it and the chief surveyor emailed it to me. The software didn't ask me to accept anything either, somewhere in its innards should be all the information required to transform coordinates from ATS77 and NAD27 (Metric and Imperial) PEI to NAD83 CSRS. I noticed that the native QGIS transformation puts the points about 2m away from where the software places them so there must be a difference in the transformations used.

**Comment by tveinot** [comment]
Wayne Tremblay <wltremblay@gov.pe.ca> our Chief Surveyor with the province has the PEI grid shift file and the software. The software is on disk but he can email the gsb if you ask. I was just speaking with him and he didn't think there is a license agreement or any issues with distribution, further indicating that they have given it to Esri for the same purpose.
Hopefully with more Atlantic Canada grids and tools available qgis will gain more footing here. 
Tyler

**Comment by gioman** [comment]
@tveinot setting up a page where to download the gdb grid and with the license (if you don't have any you can just use a Creative Commons one) would "help". I don't need anything else other than the gsb file, the EPSG codes of the CRSs supported by this grid file and possible sample data. I don't need the software, this QGIS plugin uses ogr2ogr for the underlying transformation operations.

**Comment by tveinot** [comment]
So I had a chance to get into the software I was talking about and pulled out the gridshift files it uses and it shifts coordinates to the same position as the NTV2_0.gsb file available at NRCan, NAD27 to NAD83, so I guess I don't really have anything to contribute.
However, New Brunswick does have one specific to New Brunswick and when I open the file in notepad++ the metadata there says it is newer than the NTV2_0.gsb so you might want to grab and include that one. Again it is distributed under the open government license and available for download at http://www.snb.ca/geonb1/e/DC/catalogue-E.asp, along with other things, from the geonewbrunswick data catalog. When I use it on PEI coordinates it shifts them 1m away from the NTV2_0.gsb from NRCan so there is a difference between the 2 shifts.
Well I hope that all helps.
Tyler

**Comment by gioman** [comment]
> Well I hope that all helps.

it will indeed! problem is that I would not be able (in the foreseeable future) to add transformations in my free/spare time.

**Comment by tveinot** [comment]
That is understandable, I would offer to help but I am not very good at
programming; I have a very hard time setting up the tools properly just to
try and play and learn.
Tyler

**Comment by mwtoews** [comment]
Note that this gridshift file is now part of proj-datumgrid's north-america regional package; see https://github.com/OSGeo/proj-datumgrid/issues/26

This file can also be brought into this project, but be aware that PROJ internally calls this file `ntv2_0.gsb` and not `NTV2_0.GSB` as distributed, so I'd advise changing the filename to lowercase to enable it to work well with other software on case-sensitive systems.

**Comment by gioman** [comment]
@mwtoews thanks for the heads up!
